### PR TITLE
XXZ + DM interactions and more error control

### PR DESCRIPTION
**Pull request by dominikkiese**
XXZ + DM interactions were implemented (including the respective parquet and multiloop terms). Furthermore, additional keywords were added to the launcher function, to make algorithmic precisions amenable to the user. Some tuning with respect to default parameters and scanning was performed and tested.

### CODE REVIEW

File: src/Frequency/buffer.jl [hunk 1]
```diff
@@ -1,4 +1,29 @@
...
+* `sgn_ν         :: Bool`  : sign function for second spin index 
+* `kernel        :: Int64` : specification of asymptotic kernel to be interpolated
+* `p1            :: Param` : interpolation parameters for bosonic frequency argument
+* `p2            :: Param` : interpolation parameters for first fermionic frequency argument
+* `p3            :: Param` : interpolation parameters for second fermionic frequency argument
+"""
+struct Buffer
+    exchange_flag :: Bool
+    map_flag      :: Bool 
+    sgn_μν        :: Bool
```
Comment (tobiaslcmueller) [inline src/Frequency/buffer.jl:19]: Is the combined sign really necessary, if we have the single ones?

File: src/Observable/correlation_lib/correlation_u1_dm.jl [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    bt2 = get_buffer_t(v - vp, 0.5 * (v + vp), 0.5 * (v + vp), m)
+    bu2 = get_buffer_u(0.0, vp, v, m)
+
+    # compute value
+    inner = get_Γ_comp(1, site, bs1, bt1, bu1, r, a, apply_flags_u1_dm) / (2.0 * pi)^2
+
+    if site == 1
+        vzz    = get_Γ_comp(2, site, bs2, bt2, bu2, r, a, apply_flags_u1_dm)
+        vdd    = get_Γ_comp(4, site, bs2, bt2, bu2, r, a, apply_flags_u1_dm)
+        inner += (vzz - vdd) / (2.0 * (2.0 * pi)^2)
```
Comment (tobiaslcmueller) [inline src/Observable/correlation_lib/correlation_u1_dm.jl:28]: Why is it vzz, not vxx? Maybe need to see the notes on that...

**Review by tobiaslcmueller — APPROVED** [review]
Few minor comments, nothing blocking.

Good to merge, but would spelling should be corrected ;)

File: src/Frequency/buffer.jl (hunk 1 above)
Comment (dominikkiese) [reply src/Frequency/buffer.jl:19]: I initially thought yes, there are symmetries which give directly the product xi_mu * xi_nu, whereas other only give you a factor xi_mu or xi_nu. However, I see your point, probably combined symmetries which e.g. give a combination xi_mu^2 * xi_nu can also be captured if only the single factors are saved, but I might have had a reason to it like this (been a while).

Comment (tobiaslcmueller) [reply src/Frequency/buffer.jl:19]: Let's leave it with how it is for now, will raise an issue to look into that further.

**Pull request merged**